### PR TITLE
Switch to LOG_DEBUG style log macros

### PR DESCRIPTION
**Pull request by whot**
On top of #1858, mostly filing this to check if the CI passes :)

`__VA_ARGS__` has been a thing in macros for quite a while now so the old justification for the current log macros no longer applies, we can improve them. This patch changes

```
  LOG((CLOG_WARN "some error: %d", foo));
```
to a much more natural
```
  LOG_WARN("some error: %d", foo);
```

### CODE REVIEW

**Comment by shymega** [comment]
Going to assign a review req to @p12tic given its a big PR.

File: src/lib/base/Log.cpp [hunk 1]
```diff
@@ -132,74 +133,48 @@ Log::print(const char* file, int line, const char* fmt, ...)
...
+    if (n == -1) {
+        output(kERROR, "Failed to print to log (invalid arguments)");
+        return;
     }
+
+    remaining -= n;
+    offset += n;
+
+#ifndef NDEBUG
+    n = snprintf(buffer.data() + offset, remaining - offset, "\n\t%s,%d", file, line);
```
Comment (p12tic) [inline src/lib/base/Log.cpp:161]: std::snprintf

Comment (p12tic) [reply src/lib/base/Log.cpp:161]: Fixed in another PR.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
Rebased

**Pull request merged**